### PR TITLE
Split build future into smaller futures

### PR DESCRIPTION
**Pull request by nlordell**
Currently the build future is very large, this PR splits it into 3 smaller futures and moves the build future unit tests to the `transaction::build` module closer to the code they pertain to.

This is part of an effort to split #137 into more manageable chunks.

### Test Plan

Unit tests!

### CODE REVIEW

File: src/transaction/build.rs [hunk 1]
```diff
@@ -9,226 +9,550 @@ use crate::future::{CompatCallFuture, MaybeReady};
...
-                macro_rules! maybe {
-                    ($o:expr, $c:expr) => {
-                        match $o {
-                            Some(v) => MaybeReady::ready(Ok(v)),
-                            None => MaybeReady::future($c),
-                        }
-                    };
-                }
+impl<T: Transport> BuildOfflineSignedTransactionFuture<T> {
+    /// Create a new future for building a locally singed transaction request
```
Comment (fleupold) [inline src/transaction/build.rs]: nit: typo si*gn*ed

**Review by fleupold — APPROVED** [review]
Nice, despite this adding a lot of new structs, I think it clears up what's happening and making it self-documenting. 

Thanks for going through this refactor!

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia3.giphy.com%2Fmedia%2Fd3mmkM7o2qC78iZ2%2Fgiphy.gif&token=champ" data-gitmeme-token="champ"></a>

File: src/transaction/build.rs (hunk 1 above)
Comment (nlordell) [reply src/transaction/build.rs]: Nice catch!

**Comment by nlordell** [comment]
I think it will make #137 easier to review. Also splitting that PR into this one and #142 helped make the abstractions a bit clearer.

**Pull request merged**